src/datasets/__init__.py

```diff
@@ -21,10 +21,11 @@
 __version__ = "1.11.1.dev0"
 
 import pyarrow
+from packaging import version as _version
 from pyarrow import total_allocated_bytes
 
 
-if tuple(int(i) for i in pyarrow.__version__.split(".")[:3]) < (1, 0, 0):
+if _version.parse(pyarrow.__version__).major < 1:
```

Comment:
I wonder why we make this validation, as `pyarrow` least version is already pinned in setup.py `install_requires`.

Reply:
Because in certain environments (google colab, kaggle kernels), it may create some issues.

For example on colab, pyarrow is already installed. When colab only had pyarrow 0.0.6, installing `datasets` would install pyarrow >= 1.0.0 BUT it would also require the kernel to be restarted for the update to take effect.
Because of that you could end up with an old version of pyarrow until your restart the kernel